tests/integration.rs

```diff
@@ -153,72 +155,438 @@ mod integration_tests {
+        let collector = FakeCollectorServer::start()
+            .await
+            .expect("fake collector server should start");
+        let collector_endpoint = collector.endpoint().clone();
+
+        tokio::task::spawn_blocking(|| {
+            run_test_inited(
+                "wasi-observe-tracing",
+                SpinConfig {
+                    binary_path: spin_binary(),
+                    spin_up_args: Vec::new(),
+                    app_type: SpinAppType::Http,
+                },
+                ServicesConfig::none(),
+                |env| {
+                    env.set_env_var("OTEL_EXPORTER_OTLP_ENDPOINT", collector_endpoint);
+                    env.set_env_var("OTEL_EXPORTER_OTLP_TRACES_PROTOCOL", "grpc");
+                    env.set_env_var("OTEL_BSP_SCHEDULE_DELAY", "5");
+                    Ok(())
+                },
+                move |env| {
+                    let spin = env.runtime_mut();
+                    assert_spin_request(
+                        spin,
+                        Request::new(Method::Get, "/events"),
+                        Response::new(200),
+                    )?;
+
+                    let mut spans: Vec<ExportedSpan>;
+                    assert_eventually!(
+                        {
+                            spans = collector.exported_spans();
+                            !spans.is_empty()
+                        },
+                        5
+                    );
+
+                    assert_eq!(spans.len(), 3);
+
+                    let event_span = spans
+                        .iter()
+                        .find(|s| s.name == "events")
+                        .expect("'events' span should exist");
+
+                    let events = event_span.events.clone();
+                    assert_eq!(events.len(), 3);
+
+                    let basic_event = events
+                        .iter()
+                        .find(|e| e.name == "basic-event")
+                        .expect("'basic' event should exist");
+                    let event_with_attributes = events
+                        .iter()
+                        .find(|e| e.name == "event-with-attributes")
+                        .expect("'event_with_attributes' event should exist");
+                    let event_with_timestamp = events
+                        .iter()
+                        .find(|e| e.name == "event-with-timestamp")
+                        .expect("'event_with_timestamp' event should exist");
+
+                    assert!(basic_event.time_unix_nano < event_with_attributes.time_unix_nano);
+                    assert_eq!(event_with_attributes.attributes.len(), 1);
+                    assert!(
+                        event_with_attributes.time_unix_nano < event_with_timestamp.time_unix_nano
+                    );
+
+                    Ok(())
+                },
+            )
+        })
+        .await??;
+
+        Ok(())
+    }
```

Comment:
@lann I'm in a bit of a sync/async pickle and need some help.

This test is async so that we can start the fake opentelemetry collector. But then we execute `run_test_inited` in a sync context b/c otherwise it fails. The problem is that I need to[^1] depend on a later version of the fake opentelemetry collector which makes the `exported_spans()` method async. Is there a way I can have an async context within the sync context within the async context? Is there a way I can simplify this?

[^1]: This was necessary to get access to the event fields https://github.com/davidB/tracing-opentelemetry-instrumentation-sdk/pull/160.